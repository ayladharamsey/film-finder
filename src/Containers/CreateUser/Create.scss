$background-color: white;
$margin: 5px; 
$text-color: black;
$paddingInterior: 5px;
$button-color: #c71a1a;
$box-shadow: 0px 5px 5px 0px #e8e8e8, 0px 10px 10px 0px #e8e8e8;
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:700&display=swap');
$logo-font:'Source Sans Pro', sans-serif;
$shadow-hover: #e8e8e8 - 10; 

.form {
  height: 100vh;
  display: flex;
}

.title {
  width:30%;
  justify-content: center; 
  background-color: $background-color;
  color: $button-color;
  font-family: 'Source Sans Pro', sans-serif; 
  padding: 4%;
}

h1 {
  font-size: 3.5em;
}

.sign_in-btn {
  font-family: 'Source Sans Pro', sans-serif; 
  background: $background-color;
  color: $text-color;
  width:20%;
  position: absolute;
  top: 5%;
  left: 74%;
  border: 1px solid  #f5f5f5;
  padding: $paddingInterior;
  box-shadow: 0px 5px 5px 0px #e8e8e8, 0px 10px 10px 0px #e8e8e8;
    &:hover {
      color: $background-color;
      background-color: $button-color - 10;
      box-shadow: 0px 5px 5px 0px $shadow-hover, 0px 10px 10px 0px $shadow-hover;
      border: 0px;
    }
}
.input_title {
  font-family: 'Source Sans Pro', sans-serif; 
  background-color: $background-color; 
  margin: $margin;
  padding: $paddingInterior;
  color: $text-color;
}

.input_field {
  display: flex; 
  flex-direction: column;
  justify-content: center;
  border: 1px solid $background-color;
  padding: 25px;
  background-color: $background-color; 
  width: 40%;
  float: right;
  margin: 10%;
  height: 50%;
  position: relative;
}

.name {
  background-color: $background-color; 
  margin: $margin;
  padding: $paddingInterior;
}

.email {
  background-color: $background-color;
  margin: $margin; 
  padding:$paddingInterior;
}

.password {
  background-color: $background-color; 
  margin: $margin;
  padding:$paddingInterior;
}



.button { 
  margin: $margin;
  color: $background-color;
  padding:$paddingInterior;
  background-color: $button-color;
  box-shadow: $box-shadow;
  font-weight: 900;
  box-shadow: 0px 5px 5px 0px #e8e8e8, 0px 10px 10px 0px #e8e8e8;
    &:hover {
      background-color: $button-color - 10;
      box-shadow: 0px 5px 5px 0px $shadow-hover, 0px 10px 10px 0px $shadow-hover;
    }
}

