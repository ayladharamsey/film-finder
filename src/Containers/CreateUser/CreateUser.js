import React, { Component } from 'react';
import './Create.scss';
import { setUser } from '../../actions'
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export class CreateUser extends Component { 
  constructor() { 
    super(); 
    this.state = {
      name: '',
      email: '', 
      password: '' 
    }
  }

  updateUser = e => { 
    this.setState({ [e.target.name]: e.target.value});
  }
  
    handleForm = e => { 
      e.preventDefault(); 
      this.props.addUser({
        name: this.state.name, 
        email: this.state.email, 
        password: this.state.password
      })
      this.clearInputs();
    }

    clearInputs = () => {
      this.setState({
        name: '', 
        email: '', 
        password: ''
      }); 
    }

    render() {
      const { name, email, password } = this.state; 
      return (
        <form className="form">
          <h1 className="title">FILMFINDER</h1>
          <section className="input_field">
            <Link to='/login'>
              <button className="sign_in-btn">login</button>
            </Link>
            <h2 className="input_title" >Sign Up Now!</h2>
            <input 
              className="name"
              name="name"
              value={name}
              placeholder="type your name"
              onChange={this.updateUser}
            />
            <input 
              className="email"
              name="email"
              value={email}
              placeholder="type your email"
              onChange={this.updateUser}
            />
            <input 
              className="password"
              name="password"
              value={password}
              placeholder="type your password"
              onChange={this.updateUser}
            />
            <Link to="/movies">
              <button className="button" onClick={this.handleForm}>sign up</button>
            </Link>
          </section>
        </form>
      )
    }

}

export const mapDispatchToProps = dispatch => ({
  setUser: user => dispatch(setUser(user))
});

export default connect(null, mapDispatchToProps)(CreateUser);
