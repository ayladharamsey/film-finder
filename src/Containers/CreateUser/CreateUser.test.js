import React from 'react';
import { shallow } from 'enzyme';
import { CreateUser } from './CreateUser';

describe('CreateUser container tests', () => {

  it('CreateUser should match its snapshot', () => {
    const wrapper = shallow( <CreateUser /> )
    expect(wrapper).toMatchSnapshot();
  })
})