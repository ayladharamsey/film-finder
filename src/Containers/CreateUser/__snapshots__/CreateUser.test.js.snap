// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUser container tests CreateUser should match its snapshot 1`] = `ShallowWrapper {}`;
